extensions/brand-fetch/src/index.tsx

```diff
@@ -53,8 +63,60 @@ export default function Command(props: LaunchProps<{ arguments: Arguments.Index
   }
 
   useEffect(() => {
-    handleSubmit(props.arguments.domain);
-  }, [props.arguments.domain]);
+    handleSubmit(props.arguments.query);
+  }, [props.arguments.query]);
+
+  return queryType === "domain" ? (
+    <BrandView bfApiKey={bfApiKey} domain={props.arguments.query} />
+  ) : (
+    <List isLoading={isLoading}>
+      {(resp as SearchBrandKeyword[])?.map((brand) => (
+        <List.Item
+          key={brand.brandId}
+          title={brand.name || brand.domain}
+          subtitle={brand.domain}
+          icon={brand.icon || Icon.Warning}
+          accessories={[{ icon: Icon.ChevronRightSmall }]}
+          actions={
+            <ActionPanel>
+              <Action.Push title="Preview" target={<BrandView bfApiKey={bfApiKey} domain={brand.domain} />} />
+              <Action.OpenInBrowser url={`https://brandfetch.com/${brand.domain}`} />
+            </ActionPanel>
+          }
+        />
+      ))}
+    </List>
+  );
+}
+
+function BrandView({ domain, bfApiKey }: { domain: string; bfApiKey: string }) {
+  const [data, setData] = useState<RetrieveBrandResponse | null>(null);
+  const [isLoading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong",
+        message: error,
+      });
+    }
+  }, [error]);
+
+  async function handleSubmit(input: string) {
+    try {
+      const resp = await retrieveBrand(bfApiKey, input);
+      setData(resp);
+    } catch (err: unknown) {
+      setError((err as Error).message);
```

Comment:
Worth using a [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) instead of throwing errors.